src/test/java/edu/tamu/eider/ApiDocumentation.java

```diff
@@ -0,0 +1,37 @@
+package edu.tamu.eider;
+
+import static org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.documentationConfiguration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.restdocs.RestDocumentationContextProvider;
+import org.springframework.restdocs.RestDocumentationExtension;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest
+@ExtendWith({ RestDocumentationExtension.class, SpringExtension.class })
+public class ApiDocumentation {
+
+    @Autowired
+    protected WebApplicationContext context;
```

Comment:
Building MockMvc with autowired context skips web security. Therefor; POST, PUT, DELETE tests are not requiring basic authentication.